### DISCUSSION TITLE
Using fragments without specifying unique name

### DISCUSSION THREAD

**Discussion by saivanji** [urql-graphql/urql#1109] [Q&A] [answered]
In the update functions I often create fragments for using them in `readFragment` and `writeFragment`. The problem is the fragment names have to be globally unique in `graphql-tag` across the application and it becomes tricky to maintain fragment uniqueness in different files on a scale.

I'm wondering whether there is an option not to specify fragment name when using in cache updates? I've seen `fraql` addresses that issue but that library was not updated almost a year and it seems not working with `urql` anymore.

I've found that [here](https://formidable.com/open-source/urql/docs/graphcache/custom-updates/#cachewritefragment) underscore symbol is used as a fragment name, but I'm getting warning from `graphql-tag` if I use that name more than once.

**Comment by kitten** [comment, marked as answer]
Basically this restriction only exists in GraphQL Tag and what they were meaning to do is disallow fragments from being created that accidentally have the same name in the same query. However, it's completely fine to create some fragments that have the same name if they're not being used in the same query.

So the solution here would be that you could also just use `parse` from `'graphql'`. It depends on your use-case, hence we've used that placeholder name in the example snippets.

It's also possible however to call `disableFragmentWarnings` from `graphql-tag` to prevent these warnings: https://github.com/apollographql/graphql-tag/blob/fbbb02e53de7e8a9a9d849ed2ee306d4d0599bfb/src/index.js#L176

**Comment by saivanji** [comment]
Thanks, using `parse` did the trick
